modules/project/variables.tf

```diff
@@ -208,6 +208,23 @@ variable "project_create" {
   default     = true
 }
 
+variable "service_agent_encryption_key_ids" {
+  description = "Service Agents to be granted encryption/decryption permissions over Cloud KMS encryption keys. Format {SERVICE_AGENT => [KEY_ID]}."
+  type        = map(list(string))
+  default     = {}
+  nullable    = false
+}
+
+variable "service_agents_config" {
+  description = "Automatic service agent configuration options."
+  type = object({
+    create_primary_agents = optional(bool, true)
+    grant_default_roles   = optional(bool, true)
```

Comment:
What happens if I set create to false and grant to true? Do we need validation?

I'm not convinced about having grant defaulting to true, but I already know I'm in the minority. Let's discuss it briefly. :)